### PR TITLE
[Fix] Memory leak in MNE library in bem related functions

### PR DESCRIPTION
**Pull request by juangpc**
In ```add_bem_surface``` there seems to be a leak of ```newSurf``` 

In ```add_replace_display_surface``` I'm not sure about ```surf``` variable. @LorenzE do you remember about this function?

### CODE REVIEW

File: libraries/mne/c/mne_msh_display_surface_set.cpp [hunk 1]
```diff
@@ -442,21 +442,14 @@ int MneMshDisplaySurfaceSet::add_bem_surface(MneMshDisplaySurfaceSet* surfs,
...
 
 bad : {
-        if(surf)
-        {
-            delete surf;
-        }
-        if(newSurf)
-        {
-            delete newSurf;
-        }
```
Comment (LorenzE) [inline libraries/mne/c/mne_msh_display_surface_set.cpp]: Why are we not cleaning up newSurf anymore?

File: libraries/mne/c/mne_msh_display_surface_set.cpp [hunk 2]
```diff
@@ -427,7 +427,7 @@ int MneMshDisplaySurfaceSet::add_bem_surface(MneMshDisplaySurfaceSet* surfs,
         if (std::fabs(sum - 1.0) > 1e-4) {
             fprintf(stderr, "%s surface is not closed "
                                  "(sum of solid angles = %g * 4*PI).",name,sum);
-            goto bad;
+            return FAIL;
```
Comment (LorenzE) [inline libraries/mne/c/mne_msh_display_surface_set.cpp]: if we go do not go to bad here, we do not clean up newSurf.

File: libraries/mne/c/mne_msh_display_surface_set.cpp (hunk 1 above)
Comment (juangpc) [reply libraries/mne/c/mne_msh_display_surface_set.cpp]: Yes. We're on that. This  is WP. 
But before getting on with it, I'd really need some help trying to understand what is going on with ```add_repace_display_surface``` which also (probably) has a leak. I'd like to solve both issues at the same time. 
Could you help me understand what is this fcn doing?

here: https://github.com/mne-tools/mne-cpp/blob/edffebb123ce116e5bd1ef027acb8a9a217cb8c0/libraries/mne/c/mne_msh_display_surface_set.cpp#L459

Comment (LorenzE) [reply libraries/mne/c/mne_msh_display_surface_set.cpp]: I guess we parse through a set of display surfaces and if we find a match (based on the ID) we delete the old surface and replace it with a new one?

Comment (juangpc) [reply libraries/mne/c/mne_msh_display_surface_set.cpp]: Thank you.

**Comment by juangpc** [comment]
@LorenzE , ready for review.

**Review by gabrielbmotta — DISMISSED** [review]

**Review by gabrielbmotta — APPROVED** [review]

**Pull request merged**